### PR TITLE
feat: check conventional commits in drone

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
We agreed on following https://www.conventionalcommits.org/en/v1.0.0/ 
This drone step will ensure this.

### Not compliant
image

### Compliant
Screenshot from 2023-11-16 10-06-54

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -3040,6 +3040,25 @@ def checkStarlark():
         },
     }]
 
+def checkGitCommit():
```
Comment (IljaN) [inline .drone.star:3043]: Maybe rename to checkGitCommitMessage?

Comment (IljaN) [reply .drone.star:3043]: Or "checkCommitMessage"

Comment (DeepDiver1975) [reply .drone.star:3043]: I have no hard feelings here ...
@phil-davis what's your suggestion here?

Comment (phil-davis) [reply .drone.star:3043]: I think the name is OK. It is just in this .drone.star and not a really tricky thing.

**Review by phil-davis — APPROVED** [review]
Looks OK to me - let's do it here in core, and expand to other repos when we feel that it is "a good thing"

**Pull request merged**